### PR TITLE
fix: fold variable expressions before remote run

### PR DESCRIPTION
**Pull request by LeftHandCold**
## Summary
- fold session variable expressions on a per-execution remote-run encoding copy before sending scopes to remote CNs
- return a regular error instead of panicking when a remote process has no variable resolver
- add regression coverage for filter, aggregate argument, lock path, and prepared statement reuse with @@sql_mode

## Verification
- GOCACHE=/private/tmp/mo-gocache-var-expr-hotfix GOFLAGS=-mod=mod go test ./pkg/sql/colexec ./pkg/sql/compile -run 'TestVarExpressionExecutorWithoutResolveVariableFunc|TestScopeContainsVarExpr|TestFoldVarExprs|Test_prepareRemoteRunSendingData' -count=1

### CODE REVIEW

**Review by heni02 — APPROVED** [review]

File: pkg/sql/colexec/lockop/lock_op.go [hunk 1]
```diff
@@ -1084,6 +1084,36 @@ func (lockOp *LockOp) AddLockTargetWithMode(
...
+}
+
+func (lockOp *LockOp) RewriteLockRowsExpressions(rewrite func(*plan.Expr) (*plan.Expr, bool, error)) (bool, error) {
+	folded := false
+	targets := make([]lockTarget, len(lockOp.targets))
+	copy(targets, lockOp.targets)
+	for i := range targets {
+		expr, exprFolded, err := rewrite(targets[i].lockRows)
+		if err != nil {
+			return false, err
```
[the remark this answers is not in the document]
Comment (LeftHandCold) [reply pkg/sql/colexec/lockop/lock_op.go:1107]: Fixed in `540b33b46e`: `RewriteLockRowsExpressions` now skips nil `lockRows` before invoking the callback. I also extended the hidden-expression regression test to include a nil lock target and assert the callback is only called for the non-nil expression.
